File: Modules/SMES/MES/SIE.Wpf.MES/Storages/Commands/EditStorageAreaExtCommand.cs
using SIE.Domain;
using SIE.MetaModel.View;
using SIE.Wpf.Command;

namespace SIE.Wpf.MES.Storages.Commands
{
    /// <summary>
    /// 工位货区从表修改按钮
    /// </summary>
    [Command(Label = "修改", GroupType = CommandGroupType.Edit, ImageName = "EditEntity", Location = CommandLocation.All, Gestures = "Ctrl+Shift+E")]
    public class EditStorageAreaExtCommand : ListEditCommand
    {
        /// <summary>
        /// 工位货区从表修改按钮执行方法
        /// </summary>
        /// <param name="view">实体</param>
        public override void Execute(ListLogicalView view)
        {
            base.Execute(view);
            if (view.Parent.Current.PersistenceStatus == PersistenceStatus.Unchanged)
                view.Parent.Current.PersistenceStatus = PersistenceStatus.Modified;
        }
    }
}

File: Modules/Kit/MES/SIE.Web.Kit.MES/Storages/Commands/EditStorageAreaExtCommand.js
SIE.defineCommand('SIE.Web.Kit.MES.Storages.Commands.EditStorageAreaExtCommand', {
    extend: 'SIE.cmd.Edit',      
    execute: function (view, source) {
        var me = this;
        var parent = this.view.getParent();       
        //if (parent.PersistenceStatus == 0)
        //    parent.PersistenceStatus = 1;
    }
});
